recipes/jinja2cpp/1.1.0/conanfile.py

```diff
@@ -0,0 +1,95 @@
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False], "fPIC": [True, False]
+    }
+    default_options = {'shared': False, "fPIC": True}
+    requires = (
+        "variant-lite/[>=1.2.2]",
+        "expected-lite/[>=0.3.0]",
+        "optional-lite/[>=3.2.0]",
+        "string-view-lite/[>=1.3.0]",
+        "boost/[>=1.69.0 <1.72.0]",
+#        "boost/1.70.0",
+        "fmt/[>=5.3]",
+        "rapidjson/[>=1.1.0]"
+    )
+    
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    _cpp_std = 14
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+        cppstd = self.settings.get_safe("compiler.cppstd")
+        if cppstd:
+            cppstd_pattern = re.compile(r'^(gnu)?(?P<cppstd>\d+)$')
+            m = cppstd_pattern.match(cppstd)
+            cppstd_profile = int(m.group("cppstd"))
+            if cppstd_profile < 14:
+                raise ConanInvalidConfiguration("Minimum C++ Standard required is 14 (provided '{}')".format(cppstd))
+            else:
+                self._cpp_std = cppstd_profile
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "Jinja2Cpp-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.definitions["JINJA2CPP_BUILD_TESTS"] = False
+        cmake.definitions["JINJA2CPP_STRICT_WARNINGS"] = False
+        cmake.definitions["JINJA2CPP_BUILD_SHARED"] = self.options.shared
+        cmake.definitions["JINJA2CPP_DEPS_MODE"] = "conan-build"
+        cmake.definitions["JINJA2CPP_CXX_STANDARD"] = self._cpp_std
+        # Conan cmake generator omits the build_type flag for MSVC multiconfiguration CMake,
+        # but provide build-type-specific runtime type flag. For now, Jinja2C++ build scripts
+        # need to know the build type is being built in order to setup internal flags correctly
+        cmake.definitions["JINJA2CPP_CONAN_BUILD_TYPE"] = self.settings.build_type
+        compiler = self.settings.get_safe("compiler")
+        if compiler == 'Visual Studio':
+            # Runtime type configuration for Jinja2C++ should be strictly '/MT' or '/MD'
+            runtime = self.settings.get_safe("compiler.runtime")            
+            if runtime == 'MTd':
+                runtime = 'MT'
+            if runtime == 'MDd':
+                runtime = 'MD'
+            cmake.definitions["JINJA2CPP_MSVC_RUNTIME_TYPE"] = '/' + runtime
+            
+        cmake.configure(build_folder=self._build_subfolder)
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy("*.h", dst="include", src=os.path.join(self._source_subfolder, "include"))
```

Comment:
Looking at jinja2cpp's github repo, the cmake script has install support.

Reply:
Yes, but it's too generic. It copies cmake support files, extra headers etc. which are prohibited by conan center hooks. So, I don't think it can be useful.

Reply:
Those can be removed.
But if the cmake script is not auto-generating any extra `.h` headers, then it's ok to bypass it.

Reply:
No, it is not.